tests/unit/test_resolver.py

```diff
@@ -172,43 +172,113 @@ def _resolve_packages(request: Request) -> RequestOutput:
         mock.patch.dict(
             resolver._package_managers,
-            values={"gomod": mock_resolver},
+            values={"gomod": mock.Mock(return_value=RequestOutput.empty())},
             clear=True,
         ),
         mock.patch.dict(
             resolver._dev_package_managers,
-            values={"shrubbery": mock_resolver},
+            values={},
             clear=True,
         ),
+        mock.patch(
+            "hermeto.core.models.input.get_args",
+            return_value=("bundler", "cargo", "generic", "gomod", "npm", "pip", "rpm", "yarn"),
+        ),
     ):
-        dev_package_input = mock.Mock()
-        dev_package_input.type = "shrubbery"
+        package_input = mock.Mock()
+        package_input.type = package_type
 
         request = mock.Mock()
         request.source_dir = RootedPath(tmp_path)
-        request.flags = flags
-        request.packages = [dev_package_input]
+        request.flags = []
+        request.packages = [package_input]
+
+        expected_error = re.escape(f"Package manager(s) not yet supported: {package_type}")
+        with pytest.raises(UnsupportedFeature, match=expected_error):
+            resolver.resolve_packages(request)
 
-        if flags:
-            assert resolver.resolve_packages(request) == RequestOutput(
-                components=[], build_config=BuildConfig(environment_variables=[], project_files=[])
-            )
-        else:
-            expected_error = re.escape("Package manager(s) not yet supported: shrubbery")
-            with pytest.raises(UnsupportedFeature, match=expected_error):
-                resolver.resolve_packages(request)
+
+def test_experimental_package_used_without_x_prefix_suggests_x_prefix(tmp_path: Path) -> None:
+    """Test that using experimental package without x-prefix suggests the x-prefix version."""
+    from typing import get_args
+
+    from hermeto.core.errors import InvalidInput
+    from hermeto.core.models.input import PackageManagerType, Request, parse_user_input
+
+    # Mock get_args to return the experimental type when called
+    original_args = get_args(PackageManagerType)
+    with mock.patch(
+        "hermeto.core.models.input.get_args", return_value=original_args + ("x-shrubbery",)
+    ):
+        # Create input data that will trigger input validation
+        input_data = {
+            "source_dir": str(tmp_path),
+            "output_dir": str(tmp_path / "output"),
+            "packages": [
+                {"type": "shrubbery", "path": "."}
+            ],  # User uses base type instead of x-shrubbery
+        }
+
+        # Should raise InvalidInput with experimental suggestion
+        with pytest.raises(InvalidInput) as exc_info:
+            parse_user_input(Request.model_validate, input_data)
+
+        # Should suggest x-prefix since experimental is available
+        error_message = str(exc_info.value)
+        assert "Package manager(s) not yet fully supported: shrubbery" in error_message
```

Comment:
Do you need these asserts? They make your tests fragile -- the moment someone updates the message they will have to fix tests. WDYT about introducing a more specific exception class just for signaling this problem? It should be a subclass of `InvalidInput` and that would free you to modify the message as you wish while verifying that it was actually displayed.